### PR TITLE
Delete from list funtionalities

### LINKED ISSUE #82: Delete from list
As a user, I want to be able to delete knots from the list.
As a user, I want to be able to delete intents from the list.

### PR DESCRIPTION
**Pull request by veikkahonkanen**
Resolves #82

### CODE REVIEW

File: src/components/knot-components/knot-list/knot-list.tsx [hunk 1]
```diff
@@ -68,15 +72,29 @@ const KnotPanel: React.FC<Props> = ({ knots, onKnotClick }) => {
...
+                return (
+                  <InteractiveListItem
+                    icon={ <KnotIcon htmlColor="#000"/> }
+                    title={ knot.name }
+                    onClick={ () => onKnotClick(knot) }
+                    onSecondaryActionClick={ () => onKnotSecondaryClick(knot) }
+                    secondaryActionIcon={ <DeleteOutlineIcon htmlColor="#000"/> }
+                  />
+                );
+              }
```
Comment (Chi-Chu319) [inline src/components/knot-components/knot-list/knot-list.tsx]:
```suggestion
              return KnotScope.Home? {
                  <InteractiveListItem
                    icon={ <HomeIcon htmlColor="#000"/> }
                    title={ knot.name }
                    onClick={ () => onKnotClick(knot) }
                  />
                } : {
                  <InteractiveListItem
                    icon={ <KnotIcon htmlColor="#000"/> }
                    title={ knot.name }
                    onClick={ () => onKnotClick(knot) }
                    onSecondaryActionClick={ () => onKnotSecondaryClick(knot) }
                    secondaryActionIcon={ <DeleteOutlineIcon htmlColor="#000"/> }
                  />
                }
```

File: src/components/views/story-editor-view/story-editor-view.tsx [hunk 2]
```diff
@@ -111,18 +115,45 @@ const StoryEditorView: React.FC<Props> = ({
...
-    const engine = engineRef.current;
-    const nodes = engine.getModel().getNodes();
+    if(!deletedKnot || !deletedKnot.id) {
+      return;
+    }
+
+    const model = engineRef.current.getModel();
+    const node = model.getNodes().find(item => item.getID() === deletedKnot?.id);
+    if (node) {
+      console.log(node)
```
Comment (Chi-Chu319) [inline src/components/views/story-editor-view/story-editor-view.tsx]:
```suggestion
```

File: src/components/views/story-editor-view/story-editor-view.tsx [hunk 3]
```diff
@@ -111,18 +115,45 @@ const StoryEditorView: React.FC<Props> = ({
...
+    if(!deletedIntent || !deletedIntent.id) {
+      return;
     }
-    if (knots[1] && knots[1].scope !== KnotScope.Home) {
-      knots[1].scope = KnotScope.Home;
+
+    const model = engineRef.current.getModel();
+    const link = model.getLinks().find(item => item.getID() === deletedIntent?.id);
+    if (link) {
+      console.log(link)
```
Comment (Chi-Chu319) [inline src/components/views/story-editor-view/story-editor-view.tsx]:
```suggestion
```

**Review by Chi-Chu319 — CHANGES_REQUESTED** [review]
Changes requested

File: src/components/generic/generic-dialog/generic-dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  fullScreen?: boolean;
+  fullWidth?: boolean;
+  disableEnforceFocus?: boolean;
+  disabled?: boolean;
+}
+
+/**
+ * React component displaying confirm dialogs
+ */
+const GenericDialog: React.FC<Props> = ({
```
Comment (lauriLukkarinen) [inline src/components/generic/generic-dialog/generic-dialog.tsx]:
```suggestion
/**
 * React component displaying confirm dialogs
 *
 * @param props component properties
 */
const GenericDialog: React.FC<Props> = ({
```

File: src/components/generic/list-items/interactive-list-item.tsx [hunk 5]
```diff
@@ -25,6 +33,16 @@ const InteractiveListItem: React.FC<Props> = ({ icon, title, onClick }) => {
       <ListItemText>
         { title }
       </ListItemText>
+      { (onSecondaryActionClick && secondaryActionIcon) &&
```
Comment (lauriLukkarinen) [inline src/components/generic/list-items/interactive-list-item.tsx]:
```suggestion
      { onSecondaryActionClick && secondaryActionIcon &&
```

File: src/components/knot-components/knot-list/knot-list.tsx [hunk 6]
```diff
@@ -68,15 +72,24 @@ const KnotPanel: React.FC<Props> = ({ knots, onKnotClick }) => {
...
+                /> : 
+                <InteractiveListItem
+                  icon={ <KnotIcon htmlColor="#000"/> }
+                  title={ knot.name }
+                  onClick={ () => onKnotClick(knot) }
+                  onSecondaryActionClick={ () => onKnotSecondaryClick(knot) }
+                  secondaryActionIcon={ <DeleteOutlineIcon htmlColor="#000"/> }
+                />
+              
+            })
```
Comment (lauriLukkarinen) [inline src/components/knot-components/knot-list/knot-list.tsx]:
```suggestion
            knots.map(knot =>
              <InteractiveListItem
                icon={ <KnotIcon htmlColor="#000"/> }
                title={ knot.name }
                onClick={ () => onKnotClick(knot) }
                onSecondaryActionClick={
                  knot.scope === KnotScope.Home ?
                    () => onKnotSecondaryClick(knot) :
                    undefined
                }
                secondaryActionIcon={
                  knot.scope === KnotScope.Home ?
                    <DeleteOutlineIcon htmlColor="#000"/> :
                    undefined
                }
              />
            )
```

File: src/components/screens/editor/editor-screen.tsx [hunk 7]
```diff
@@ -369,6 +375,54 @@ const EditorScreen: React.FC<Props> = ({
...
+        selectedKnot: undefined
+      }) :
+      setStoryData({
+        ...storyData,
+        selectedKnot: entity as Knot,
+        selectedIntent: undefined
+      });
+
+    setDeleteConfirmDialogOpen(true);
+  }
```
Comment (lauriLukkarinen) [inline src/components/screens/editor/editor-screen.tsx]:
```suggestion
  const onDeleteFromListClick = (entity: Knot | Intent) => {
    setStoryData({
      ...storyData,
      selectedIntent: (entity as Intent).sourceKnotId ?
        entity as Intent :
        entity as Knot,
      selectedKnot: undefined
    });

    setDeleteConfirmDialogOpen(true);
  }
```

File: src/components/screens/editor/editor-screen.tsx [hunk 8]
```diff
@@ -800,6 +858,30 @@ const EditorScreen: React.FC<Props> = ({
...
+    return (
+      <GenericDialog
+        title={ strings.editorScreen.confirm.title }
+        positiveButtonText={ strings.generic.remove }
+        cancelButtonText={ strings.generic.cancel }
+        onClose={ () => setDeleteConfirmDialogOpen(false) }
+        onCancel={ () => setDeleteConfirmDialogOpen(false) }
+        onConfirm={ onDeleteConfirmClick }
+        open= { deleteConfirmDialogOpen }
+        error= { false }
```
Comment (lauriLukkarinen) [inline src/components/screens/editor/editor-screen.tsx]:
```suggestion
        open={ deleteConfirmDialogOpen }
        error={ false }
```

**Review by lauriLukkarinen — APPROVED** [review]

**Pull request merged**